XenAdmin/MainWindow.cs

```diff
@@ -3028,21 +3028,26 @@ private void navigationPane_TreeViewSelectionChanged()
 
         private void navigationPane_NotificationsSubModeChanged(NotificationsSubModeItem submodeItem)
         {
-            alertPage.Visible = submodeItem.SubMode == NotificationsSubMode.Alerts;
-            updatesPage.Visible = submodeItem.SubMode == NotificationsSubMode.Updates;
-            eventsPage.Visible = submodeItem.SubMode == NotificationsSubMode.Events;
-            TheTabControl.Visible = false;
-
-            if (alertPage.Visible)
-                alertPage.RefreshAlertList();
-
-            if (updatesPage.Visible)
-                updatesPage.RefreshUpdateList();
-
-            if (eventsPage.Visible)
+            switch (submodeItem.SubMode)
             {
-                eventsPage.RefreshDisplayedEvents();
-            }
+                case NotificationsSubMode.Alerts:
+                    alertPage.ShowPage();
+                    updatesPage.HidePage();

```

Comment:
Before calling HidePage, do we need to check for page visibilty as we do in navigationPane_NavigationModeChanged? Because only one of the other two pages will be visible at a time; if, for example, the visible page was the updatePage, we can thus avoid the iteration through the actions when hiding the eventsPage which was invisible.